src/NavList/NavList.tsx

```diff
@@ -26,32 +30,55 @@ export type NavListItemProps = {
   children: React.ReactNode
   href?: string
   'aria-current'?: 'page' | 'step' | 'location' | 'date' | 'time' | 'true' | 'false' | boolean
-  // sx
 }
 
+const ItemContext = React.createContext<{depth: number}>({depth: 0})
+
+// TODO: sx prop
+// TODO: as prop
 const Item = React.forwardRef<HTMLAnchorElement, NavListItemProps>(
   ({href, 'aria-current': ariaCurrent, children}, ref) => {
+    const {depth} = React.useContext(ItemContext)
+
+    // TODO: Test this error case
+    if (depth > 1) {
```

Comment:
This was a bit confusing to understand inside Item, would it make more sense to check this in SubNav?

If `SubNav` has `SubNavContext`, that means it was nested twice, flag it! We don't need to check it with ItemContext 🤔 

```jsx
<NavList>
  <NavList.Item href="#">Item 1</NavList.Item>
  <NavList.Item>
    Item 2
    <NavList.SubNav>
      <NavList.Item href="#" aria-current="page">
        Sub item 1
      </NavList.Item>
      <NavList.Item href="#">
        Sub Item 2
-->     <NavList.SubNav>
            <NavList.Item href="#" aria-current="page">double sub item 1</NavList.Item>
            <NavList.Item href="#" aria-current="page">double sub item 2</NavList.Item>
        </NavList.SubNav>
      </NavList.Item>
    </NavList.SubNav>
  </NavList.Item>
  <NavList.Item href="#">Item 3</NavList.Item>
</NavList>
```

Reply:
We need to check the depth here to change the indentation of the item: https://github.com/primer/react/pull/2064/files#diff-874836517a58028b99e00e884887b49d9c859a39d2fd2dc0f0a02eb679441a1dR79-R81

I'm not sure if there's a way to do that from SubNav